src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1560,6 +1560,16 @@ LedgerManagerImpl::applyTransactions(
             storeTransaction(mApp.getDatabase(), ledgerSeq, tx, tm.getXDR(),
                              txResultSet);
         }
+        else if (mApp.getConfig().MODE_STORES_HISTORY_MINIMAL)
+        {
+            auto ledgerSeq = ltx.loadHeader().current().ledgerSeq;
+            // Clear the meta as we don't want to store it.
+            tm.clearOperationMetas();
+            tm.clearTxChangesAfter();
+            releaseAssert(xdr::xdr_size(tm.getXDR()) == 0);
+            storeTransaction(mApp.getDatabase(), ledgerSeq, tx, tm.getXDR(),
```

Comment:
Meta is pretty expensive, so it seems wasteful to first write it then immediately override it. I think we should write default-initialized TransactionMeta right away to avoid performance overhead.

Reply:
Cool yeah that makes sense. I changed it to use a minimally initialized meta (with the ledger version). We still want the `tm` to be populated prior to this for streaming, right?